### PR TITLE
port label tests with sagemaker to v2

### PR DESCRIPTION
**Pull request by sirutBuasai**
## Purpose

- Porting sagemaker label tests from v1 to v2.
    - This is split into 2 test files: 1) `inspect_docker.py` does a binary URI visibility check. This file is generally for anything docker metadata related via docker inspect. 2) `check_labels.py` specifically checks for the labels of the RC containers. This specficially checks that the container are labelled correctly for sagemaker containers to ensure sagemaker training/inference functionality. These two tests are split for organization purposes.
- Change auto-release file name and job names to `auto-release-*`. This align with the same convention we are using for `pr-*` workflows. This is so developers can easily distinguish the workflow type based on the file name at a glance.
- Splitting jobs into modules such that PR and Release workflow can consume them. The reduces code redundancy overall while providing a module for each PR and Release workflow. This split is done from a previously coupled PR and Release workflow via PR reusable workflow for the following trade offs.
```
Option A: Shared reusable workflows, composed independently
PR:      build → sanity + security + ec2
Release: build → sanity + ec2 + release

Pros
Each workflow declares exactly what it runs — no hidden skips
Easy to reason about and extend (nightly, hotfix, etc.)
Job graphs are explicit — no conditional if: chains

Cons
Duplicated with: wiring when calling shared reusable workflows
Adding a new reusable test means updating both PR and release files

Option B: PR workflow is reusable, release wraps it
PR:      build → sanity + security + ec2
Release: reusable-PR(skip_check_changes) → release

Pros
Single source of truth for build+test — change once, both get it
Less total YAML

Cons
Security skip is encoded as the absence of || needs.check-changes.result == 'skipped' in one job's if: — every other job has it. Easy to break if someone adds it for consistency without realizing it controls release behavior.
PR-specific jobs (gatekeeper, check-changes) show as "skipped" in release UI
Secrets must be explicitly passed at each nesting hop — doesn't propagate transitively
```

## Test Result

- [1136977](https://github.com/aws/deep-learning-containers/pull/5708/commits/1136977a38c857219a6cd70c8d4ca61d0bc718bb) Run all pipelines for PR and Release workflows using modularized test format
    - sanity / sagemaker-only tests skips are expected for non sagemaker images.
    - step2/3 for release workflows in gamma skips are expected due to block day.
- [43a9e79](https://github.com/aws/deep-learning-containers/pull/5708/commits/43a9e7905a2bc4f738f6c144b192ec6f8c7dedfd) Run only test-changes tests using modularized test format.
    - PR sanity tests are expected to fail since some of the images are not yet released to PROD. Moreover, some released prod images were released while /tmp was not emptied.
    - This test run is considered passing given that the skip logic and PROD image substitution is working as expected.

______________________________________________________________________

<details>
<summary>Toggle if you are merging into master Branch</summary>

By default, docker image builds and tests are disabled. Two ways to run builds and tests:

1. Using dlc_developer_config.toml
1. Using this PR description (currently only supported for PyTorch, TensorFlow, vllm, and base images)

<details>
<summary>How to use the helper utility for updating dlc_developer_config.toml</summary>

Assuming your remote is called `origin` (you can find out more with `git remote -v`)...

- Run default builds and tests for a particular buildspec - also commits and pushes changes to remote; Example:

`python src/prepare_dlc_dev_environment.py -b </path/to/buildspec.yml> -cp origin`

- Enable specific tests for a buildspec or set of buildspecs - also commits and pushes changes to remote; Example:

`python src/prepare_dlc_dev_environment.py -b </path/to/buildspec.yml> -t sanity_tests -cp origin`

- Restore TOML file when ready to merge

`python src/prepare_dlc_dev_environment.py -rcp origin`

**NOTE: If you are creating a PR for a new framework version, please ensure success of the local, standard, rc, and efa sagemaker tests by updating the dlc_developer_config.toml file:**

- [ ] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`
- [ ] `sagemaker_local_tests = true`

</details>

<details>
<summary>How to use PR description</summary>
Use the code block below to uncomment commands and run the PR CodeBuild jobs. There are two commands available:

- `# /buildspec <buildspec_path>`
  - e.g.: `# /buildspec pytorch/training/buildspec.yml`
  - If this line is commented out, dlc_developer_config.toml will be used.
- `# /tests <test_list>`
  - e.g.: `# /tests sanity security ec2`
  - If this line is commented out, it will run the default set of tests (same as the defaults in dlc_developer_config.toml): `sanity, security, ec2, ecs, eks, sagemaker, sagemaker-local`.

</details>

```
# /buildspec <buildspec_path>
# /tests <test_list>
```

</details>

<details>
<summary>Toggle if you are merging into main Branch</summary>

## PR Checklist

- [] I ran `pre-commit run --all-files` locally before creating this PR. (Read [DEVELOPMENT.md](https://github.com/aws/deep-learning-containers/blob/main/DEVELOPMENT.md) for details).

</details>

### CODE REVIEW

File: .github/workflows/reusable-sglang-upstream-tests.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        run: |
+          docker exec ${CONTAINER_ID} python3 -m sglang.bench_serving \
+          --backend sglang \
+          --host 127.0.0.1 --port 30000 \
+          --num-prompts 1000 \
+          --model Qwen/Qwen3-0.6B \
+          --dataset-name sharegpt \
+          --dataset-path /dataset/ShareGPT_V3_unfiltered_cleaned_split.json
+
+  upstream-test:
```
Comment (Yadan-Wei) [inline .github/workflows/reusable-sglang-upstream-tests.yml]: Can we rename the reusable test file name or this test name, otherwise we will have upstream-tests/upstream-test in workflow.

**Review by Yadan-Wei — COMMENTED** [review]
Thanks for making all these changes to make workflow neat and clean. All things are LGTM. For upstream tests, we can see if we have a better name for it.
